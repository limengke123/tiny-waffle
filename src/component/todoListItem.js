import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styles from '../style/component/todoListItem.module.scss'

class TodoListItem extends PureComponent {
    static propTypes = {
        data: PropTypes.shape({
            info: PropTypes.shape({
                text: PropTypes.string,
                date: PropTypes.string
            }),
            isComplete: PropTypes.bool,
            id: PropTypes.number
        }),
        handleDelete: PropTypes.func,
        handleComplete: PropTypes.func
    }

    static defaultProps = {
        data: {
            isComplete: false,
            info: {
                text: '',
                data: ''
            },
            id: ''
        },
        handleDelete() {},
        handleComplete() {}
    }

    onDelete = () => {
        const { handleDelete, data } = this.props
        handleDelete(data)
    }

    onComplete = () => {
        const { handleComplete, data } = this.props
        handleComplete(data)
    }

    render() {
        console.log('listItem is render')
        const { data } = this.props
        const { info, isComplete } = data
        const { date, text } = info
        const textClassName = classNames(styles.text, {
            [styles['is-complete']]: isComplete
        })
        return (
            <div>
                <div onClick={this.onComplete}>
                    <span className={textClassName}>{text}</span>
                    <span className={styles.date}>{date}</span>
                </div>
                <button type="button" onClick={this.onDelete}>
                    删除
                </button>
            </div>
        )
    }
}

export { TodoListItem }
