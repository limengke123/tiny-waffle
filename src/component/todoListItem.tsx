import React, { PureComponent } from 'react'
import classNames from 'classnames'
import styles from '../style/component/todoListItem.module.scss'

export interface TodoListItemProps {
    data: TodoItem
    handleDelete: (data: TodoListItemProps['data']) => void
    handleComplete: (data: TodoListItemProps['data']) => void
}

export interface TodoItem {
    isComplete: boolean
    id: number
    info: {
        text: string
        date: string
    }
}

export class TodoListItem extends PureComponent<TodoListItemProps> {
    static defaultProps = {
        data: {
            isComplete: false,
            info: {
                text: '',
                data: ''
            },
            id: ''
        },
        handleDelete() {},
        handleComplete() {}
    }

    onDelete = () => {
        const { handleDelete, data } = this.props
        handleDelete(data)
    }

    onComplete = () => {
        const { handleComplete, data } = this.props
        handleComplete(data)
    }

    render() {
        const { data } = this.props
        const { info, isComplete } = data
        const { date, text } = info
        const textClassName = classNames(styles.text, {
            [styles['is-complete']]: isComplete
        })
        return (
            <div>
                <div onClick={this.onComplete} className={styles.wrapper}>
                    <span className={textClassName}>{text}</span>
                    <span className={styles.date}>{date}</span>
                </div>
                <button type="button" onClick={this.onDelete}>
                    删除
                </button>
            </div>
        )
    }
}
