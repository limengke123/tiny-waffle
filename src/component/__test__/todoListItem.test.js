import React from 'react'
import { render, mount } from 'enzyme'
import { TodoListItem } from '../todoListItem'
import styles from '../../style/component/todoListItem.module.scss'

describe('todolistItem', () => {
    it('should render correctly', () => {
        const wrapper = render(<TodoListItem />)
        expect(wrapper).toMatchSnapshot()

        const data = {
            info: {
                text: '测试文字',
                date: '2019-2-13 12:24:42'
            },
            isComplete: true,
            id: 123
        }
        const wrapper1 = render(<TodoListItem data={data} />)
        expect(wrapper1).toMatchSnapshot()
    })

    it('should trigger delete when click button', () => {
        const handleDelete = jest.fn()
        const wrapper = mount(<TodoListItem handleDelete={handleDelete} />)
        wrapper.find('button').simulate('click')
        expect(handleDelete).toHaveBeenCalledTimes(1)
    })

    it('should trigger handleComplete when text is clicked', () => {
        const handleComplete = jest.fn()
        const wrapper = mount(<TodoListItem handleComplete={handleComplete} />)
        wrapper.find(`.${styles.wrapper}`).simulate('click')
        expect(handleComplete).toHaveBeenCalledTimes(1)
    })
})
