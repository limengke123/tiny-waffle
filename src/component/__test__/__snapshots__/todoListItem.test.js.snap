// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todolistItem should render correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <span
      class="text"
    />
    <span
      class="date"
    />
  </div>
  <button
    type="button"
  >
    删除
  </button>
</div>
`;

exports[`todolistItem should render correctly 2`] = `
<div>
  <div
    class="wrapper"
  >
    <span
      class="text is-complete"
    >
      测试文字
    </span>
    <span
      class="date"
    >
      2019-2-13 12:24:42
    </span>
  </div>
  <button
    type="button"
  >
    删除
  </button>
</div>
`;
